src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -295,16 +295,50 @@ def fetch_jwt_bundles(self) -> JwtBundleSet:
         responses = self._spiffe_workload_api_stub.FetchJWTBundles(
             workload_pb2.JWTBundlesRequest(), timeout=10
         )
-        jwt_bundles = {}
         res = next(responses)
-        for td, jwk_set in res.bundles.items():
-            jwt_bundles[TrustDomain(td)] = JwtBundle.parse(TrustDomain(td), jwk_set)
-
+        jwt_bundles: Dict[TrustDomain, JwtBundle] = self._create_td_jwt_bundle_dict(res)
         if not jwt_bundles:
             raise FetchJwtBundleError('JWT Bundles response is empty')
 
         return JwtBundleSet(jwt_bundles)
 
+    def watch_jwt_bundles(
+        self,
+        on_success: Callable[[JwtBundleSet], None],
+        on_error: Callable[[Exception], None],
+        retry_connect: bool = True,
+    ) -> CancelHandler:
+        """Watches for changes to the JWT bundles.
+
+        Args:
+            on_success: A Callable accepting a X509Context as argument and returning None, to be executed when a new
+                        update is fetched from the Workload API.
+
+            on_error: A Callable accepting an Exception as argument and returning None, to be executed when there is
+                      an error on the connection with the Workload API.
+
+            retry_connect: Enable retries when the connection with the Workload API returns an error. Default: True.
+
+        Returns:
+            CancelHandler: An object on which it can be called the method `cancel` to close the stream connection with
+                           the Workload API.
```

Comment:
Suggestion: call out that the client is responsible to call the cancel method to stop the connection whenever this is not required to avoid using resources when not needed.

Reply:
Thanks Glaucimar.
I've added the text Max used for the other watcher method so it is consistent.